### PR TITLE
🌱 Set controller-runtime logger for clusterctl

### LINKED ISSUE #9101: Faulty Initialisation of Cluster API using `clusterctl`
While initialising **cluster api** using `clusterctl` I am getting the following message but the initialisation completes. This was not happening when using `clusterctl v1.4.4`

```
[controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:
goroutine 1 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:24 +0x64
sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
        sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/log.go:59 +0xf0
sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).WithName(0x14000400000, {0x104285f61, 0x14})
        sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/deleg.go:147 +0x38
github.com/go-logr/logr.Logger.WithName({{0x104e3e0c8, 0x14000400000}, 0x0}, {0x104285f61?, 0x14000a02b78?})
        github.com/go-logr/logr@v1.2.4/logr.go:336 +0x48
sigs.k8s.io/controller-runtime/pkg/client.newClient(0x140025a2b40, {0x0, 0x140007fc150, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
        sigs.k8s.io/controller-runtime@v0.15.0/pkg/client/client.go:115 +0x7c
sigs.k8s.io/controller-runtime/pkg/client.New(0x0?, {0x0, 0x140007fc150, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
        sigs.k8s.io/controller-runtime@v0.15.0/pkg/client/client.go:101 +0x54
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.(*proxy).NewClient.func1()
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/proxy.go:169 +0x68
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.retryWithExponentialBackoff.func1()
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/client.go:232 +0x5c
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x14000a02eb8?)
        k8s.io/apimachinery@v0.27.2/pkg/util/wait/wait.go:145 +0x48
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0xee6b280, 0x3ff8000000000000, 0x3fb999999999999a, 0x9, 0x0}, 0x0?)
        k8s.io/apimachinery@v0.27.2/pkg/util/wait/backoff.go:461 +0x58
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.retryWithExponentialBackoff({0xee6b280, 0x3ff8000000000000, 0x3fb999999999999a, 0x9, 0x0}, 0x6?)
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/client.go:230 +0xcc
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.(*proxy).NewClient(0x14000a030f8?)
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/proxy.go:167 +0x98
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.listProviders({0x104e41d68?, 0x140002da150?}, 0x0?)
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/inventory.go:324 +0x30
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.(*inventoryClient).List.func1()
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/inventory.go:314 +0x2c
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.retryWithExponentialBackoff.func1()
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/client.go:232 +0x5c
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x14000a03188?)
        k8s.io/apimachinery@v0.27.2/pkg/util/wait/wait.go:145 +0x48
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0xee6b280, 0x3ff8000000000000, 0x3fb999999999999a, 0x9, 0x0}, 0x10285174c?)
        k8s.io/apimachinery@v0.27.2/pkg/util/wait/backoff.go:461 +0x58
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.retryWithExponentialBackoff({0xee6b280, 0x3ff8000000000000, 0x3fb999999999999a, 0x9, 0x0}, 0x14000a032c8?)
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/client.go:230 +0xcc
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.(*inventoryClient).List(0x1400081f140)
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/inventory.go:313 +0xac
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.(*providerInstaller).Validate(0x1400048f4a0)
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/installer.go:177 +0x38
sigs.k8s.io/cluster-api/cmd/clusterctl/client.(*clusterctlClient).Init(0x14000465410, {{{0x0, 0x0}, {0x0, 0x0}}, {0x16d5bf2af, 0xb}, {0x14000037fc0, 0x1, 0x1}, ...})
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/init.go:135 +0x1c4
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.runInit()
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/init.go:144 +0x1dc
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.glob..func13(0x10628bc00?, {0x140004de200?, 0x8?, 0x8?})
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/init.go:86 +0x1c
github.com/spf13/cobra.(*Command).execute(0x10628bc00, {0x140004de180, 0x8, 0x8})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x5c8
github.com/spf13/cobra.(*Command).ExecuteC(0x10628d5e0)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x35c
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.7.0/command.go:992
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.Execute()
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/root.go:105 +0x2c
main.main()
        sigs.k8s.io/cluster-api/cmd/clusterctl/main.go:27 +0x1c
```

### What did you expect to happen?

Clean initialisation

### Cluster API version

1.5.0

### Kubernetes version

NA

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind bug
One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels.

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #9101

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold
for confirmation on https://github.com/kubernetes-sigs/cluster-api/issues/9101#issuecomment-1662459268

**Review by killianmuldoon — COMMENTED** [review]
/lgtm
/approve

(Hold is in place to ensure this fixes the reported issue)

File: cmd/clusterctl/cmd/root.go [hunk 1]
```diff
@@ -163,7 +164,9 @@ func initConfig() {
 		}
 	}
 
-	logf.SetLogger(logf.NewLogger(logf.WithThreshold(verbosity)))
+	log := logf.NewLogger(logf.WithThreshold(verbosity))
```
Comment (furkatgofurov7) [inline cmd/clusterctl/cmd/root.go:167]: Nice one! (de-encapsulate to encapsulate it again later :))

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Comment by killianmuldoon** [comment]
/hold cancel

User confirmed that this fixed their issue.

File: cmd/clusterctl/cmd/root.go (hunk 1 above)
Comment (sbueringer) [reply cmd/clusterctl/cmd/root.go:167]: I just extracted the statement we had before :) (didn't really think about it)

File: cmd/clusterctl/cmd/root.go [hunk 2]
```diff
@@ -163,7 +164,9 @@ func initConfig() {
...
+	logf.SetLogger(log)
+	ctrl.SetLogger(log)
```
Comment (vincepri) [inline cmd/clusterctl/cmd/root.go:169]: Yes! the checks are working 😄

Comment (sbueringer) [reply cmd/clusterctl/cmd/root.go:169]: Unfortunately missed it until after the release :) But yup worked.

I wonder in how many binaries / controllers this "panic" will pop up :D

Comment (vincepri) [reply cmd/clusterctl/cmd/root.go:169]: True, but it's doing what it's supposed to do 😄

**Pull request merged**